src/routes/demographics/survey-tools.js

```diff
@@ -97,3 +106,92 @@ export function createFieldValidator(...validators) {
 
     return [{ subscribe }, action];
 }
+
+function formatIncomeInput(event) {
+  const r = /^\d/;
+  return event.target.value.split('').filter(a => r.test(a)).join('');
+}
+
+function formatZipcodeInput(event) {
+  const r = /^\d/;
+  return event.target.value.split('').filter(a => r.test(a)).join('').slice(0,5);
+}
+
+/**
+ * Formatters for free-text questions that require (1) formatting and (2) validating the values entred in,
+ * usually on blur.
+ */
+export const inputFormatters = {
+  validateAllQuestions(schema, answers) {
+    return Object.keys(schema).every(key => {
+      if (!this.hasValidator(key)) return true;
+      return this.hasValidator(key) && !this[key].isInvalid(answers[key]);
+    })
+  },
+  exists(key) {
+    return (key in this);
+  },
+  has(key, functionName) {
+    return (this.exists(key) && functionName in this[key])
+  },
+  hasValidator(key) {
+    return this.has(key, 'isInvalid');
+  },
+  showErrors(key) {
+    return this[key].showValidationErrors;
+  },
+  hasInput(key) {
+    return this.has(key, 'input');
+  },
+  hasBlur(key) {
+    return this.has(key, 'blur');
+  },
+  hasFocus(key) {
+    return this.has(key, 'focus');
+  },
+  // the individual question validators
+  zipcode: {
+    alignRight: false,
+    showValidationErrors: false,
+    input: formatZipcodeInput,
+    blur(event) {
+      const value = formatZipcodeInput(event);
+      this.showValidationErrors = true;
+      return value;
+    },
+    focus(event) {
+      const value = formatZipcodeInput(event);
+      this.showValidationErrors = false;
+      return value;
+    },
+    isInvalid(text) {
+      const number = Number(text);
+      if (text.length > 0 && text.length < 5) return 'zipcode must be a five-digit number';
```

Comment:
I'm less worried about the missing brace if it's on the same line like this. The concern is just that someone will come along and add a line to it without realizing the missing brace (the cause of the "[goto fail](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/)" bug, maybe most famously)